ui/src/layout/ActivityPanel.js

```diff
@@ -81,7 +114,28 @@ const ActivityPanel = () => {
           dispatch(scanStatusUpdate(data.scanStatus))
         }
       })
-  }, [dispatch])
+    if (open) {
+      fetchUtils
+        .fetchJson(subsonic.url('getNowPlaying'))
+        .then((resp) => resp.json['subsonic-response'])
+        .then((data) => {
+          if (data.status === 'ok') {
+            setNowPlaying(
+              data?.nowPlaying?.entry?.map((user) => {
+                return {
+                  username: user.username,
+                  coverArtId: user.coverArt,
+                  title: user.title,
+                  album: user.album,
+                  artist: user.artist,
+                  minutesAgo: user.minutesAgo || 0,
+                }
+              }) || []
+            )
+          }
+        })
+    }
+  }, [dispatch, open])
```

Comment:
the useEffect runs whenever open changes it state where open = Boolean(anchorEl) i.e whenever we either open/close the modal but we don't want to make an API call when we close the modal hence this check condtion.